logictest: run SQLLite logic tests with 3node-tenant

asubiotto opened the pull request:
We used to run this configuration as part of the default configurations, but
since the change to CCL, 3node-tenant is no longer a default configuration.

This commit adds TestTenantSQLLiteLogic, which runs the SQLLite logic tests
with a CCL binary and the 3node-tenant configuration. This test is also added
to build/teamcity-sqllogictest.sh to run nightly.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52677)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52677#-:-MEYFsRb0s4YmTdVH8jA:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52677#-:-MEYRGL62fuE0adgdRwk:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->

Commit:
logictest: run SQLLite logic tests with 3node-tenant

We used to run this configuration as part of the default configurations, but
since the change to CCL, 3node-tenant is no longer a default configuration.

This commit adds TestTenantSQLLiteLogic, which runs the SQLLite logic tests
with a CCL binary and the 3node-tenant configuration. This test is also added
to build/teamcity-sqllogictest.sh to run nightly.

Release note: None

asubiotto commented:
bors r=yuzefovich,nvanbenschoten
